src/DataTransferObject.php

```diff
@@ -7,23 +7,72 @@
 use Spatie\DataTransferObject\Attributes\CastWith;
 use Spatie\DataTransferObject\Attributes\MapTo;
 use Spatie\DataTransferObject\Casters\DataTransferObjectCaster;
+use Spatie\DataTransferObject\Descriptors\ClassDescriptor;
 use Spatie\DataTransferObject\Exceptions\UnknownProperties;
 use Spatie\DataTransferObject\Reflection\DataTransferObjectClass;
+use Spatie\DataTransferObject\Resolvers\CastResolver;
+use Spatie\DataTransferObject\Resolvers\MapFromResolver;
 
 #[CastWith(DataTransferObjectCaster::class)]
 abstract class DataTransferObject
 {
+    private static ClassDescriptor $description;
+
     protected array $exceptKeys = [];
 
     protected array $onlyKeys = [];
 
-    public function __construct(...$args)
+    public static function new(...$args): static
     {
+        $dataTransferObject = new static();
+
+        $class = new DataTransferObjectClass($dataTransferObject);
+
+        return $dataTransferObject
+            ->setUp($class, ...$args)
+            ->validate($class);
+    }
+
+    public static function newWithoutValidation(...$args): static
+    {
+        $dataTransferObject = new static();
+
         if (is_array($args[0] ?? null)) {
             $args = $args[0];
         }
 
-        $class = new DataTransferObjectClass($this);
+        $properties = (new MapFromResolver(
+            static::describe()
+        ))->mapArguments($args);
+
+        $properties = (new CastResolver(
+            static::describe()
+        ))->castArguments($properties);
+
+        return new static($properties);
+    }
+
+    private static function describe(): ClassDescriptor
+    {
+        if (! isset(static::$description)) {
+            static::$description = new ClassDescriptor(static::class);
+        }
+
+        return static::$description;
+    }
```

Comment:
This method won't work in the long run. We probably need to leave the `__construct()` method empty, have a property value resolver, and make this method non-static.

Reply:
Can you explain why it won't work?

Reply:
In the following code:

```php
class Dto1 extends DataTransferObject
{
    public string $property;
}

class Dto2 extends DataTransferObject
{
    public string $otherProperty;
}

$dto1 = new Dto1(property: 'test');

$dto2 = new Dto2(otherProperty: 'some other test');
```

Both `$dto1::description` and `$dto2::description` equal the `ClassDescriptor` created with `Dto2::class` because it is static.